integration_tests/src/main/python/json_fuzzing_test.py

```diff
@@ -0,0 +1,413 @@
+    HEX -> DIGIT 
+         | 'a' .. 'f' 
+         | 'A' .. 'F'
+    """
+    path = random.randint(0, 2)
+    if path == 0:
+        for t in gen_digit():
+            yield t
+    elif path == 1:
+        yield chr(random.randint(0x41, 0x46))
+    else:
+        yield chr(random.randint(0x61, 0x66))
+
+def gen_number():
+    """
+    NUMBER -> INTEGER FRACTION EXPONENT
+    """
+    for t in gen_integer():
+        yield t 
+    for t in gen_fraction():
+        yield t 
+    for t in gen_exponent():
+        yield t 
+
+def gen_integer():
+    """
+    INTEGER -> DIGIT 
+             | ONENINE DIGITS 
+             | '-' DIGIT 
+             | '-' ONENINE DIGITS
+    """
+    if random.randint(1, 100) <= 50:
+        yield '-'
+    
+    if random.randint(1, 100) <= 50:
+        for t in gen_digit():
+            yield t 
+    else:
+        for t in gen_onenine():
+            yield t 
+        for t in gen_digits():
+            yield t 
+
+def gen_digits():
+    """
+    DIGITS -> DIGIT 
+            | DIGIT DIGITS
+    """
+    for t in gen_digit():
+        yield t 
+
+    if random.randint(1, 100) < 70:
+        for t in gen_digits():
+            yield t 
+
+def gen_digit():
+    """
+    DIGIT -> '0' 
+           | ONENINE
+    """
+    if random.randint(0, 9) == 0:
+        yield '0'
+    else:
+        for t in gen_onenine():
+            yield t
+
+def gen_onenine():
+    """
+    ONENINE -> '1' .. '9'
+    """
+    yield chr(random.randint(0x31, 0x39))
+
+def gen_fraction():
+    """
+    FRACTION -> "" | '.' DIGITS
+    """
+    if random.randint(1, 100) < 50:
+        yield ""
+    else:
+        yield '.'
+        for t in gen_digits():
+            yield t 
+
+def gen_exponent():
+    """
+    EXPONENT -> "" 
+              | 'E' SIGN DIGITS 
+              | 'e' SIGN DIGITS
+    """
+    if random.randint(1, 100) < 20:
+        yield ""
+    else:
+        yield random.choice(['E', 'e'])
+        for t in gen_sign():
+            yield t 
+        for t in gen_digits():
+            yield t 
+
+def gen_sign():
+    """
+    SIGN -> "" 
+          | '+' 
+          | '-'
+    """
+    yield random.choice(["", '+', '-'])
+
+def gen_whitespace():
+    """
+    WHITESPACE -> '' 
+                | 0x0020 WHITESPACE 
+                | 0x000a WHITESPACE (todo)
+                | 0x000d WHITESPACE (todo)
+                | 0x0009 WHITESPACE (todo)
+    """
+    if random.randint(0, 4) > 3:
+        yield chr(random.choice([0x0020]))
+        for t in gen_whitespace():
+            yield t
+    else:
+        yield ''
+
+def gen_bool():
+    """
+    BOOL -> "true" 
+          | "null" 
+          | "false"
+    """
+    yield random.choice(["true", "null", "false"])
+
+
+
+_enable_all_types_conf = {
+    'spark.rapids.sql.format.json.enabled': 'true',
+    'spark.rapids.sql.format.json.read.enabled': 'true'}
+
+@approximate_float
+@allow_non_gpu('FileSourceScanExec')
+@pytest.mark.xfail(reason="fuzz test may randomly fail")
```

Comment:
To be clear fuzz testing is intended to find issues and should not be running by default as a part of integration tests. It should be something that we can enable and run for a configurable number of iterations to find problems. Once we find the problem we need to have a way to debug/reproduce this issue. Then we can file a follow on issue/bug to fix it. 

I would much rather have us skip this test by default unless a flag is set to enable fuzz testing, preferably with a number of iterations.  With xfail we want to get to the point where if a test passes when we expect it to fail then it will be marked as a failure, `xfail_strict=true`. Because this is totally random I really don't want to mark it as xfail.

Reply:
Done. Fuzz tests are skipped by defualt.
We can enable fuzz tests and store the tested data by running
```
> bash run_pyspark_from_build.sh -k test_json_read_fuzz --fuzz_test --debug_tmp_path
```